### DISCUSSION TITLE
Why library does not unitize Sequence collection? It falls in OutOfMemory sometimes

### DISCUSSION THREAD

**Discussion by andriishpek-sigma** [LemonAppDev/konsist#1026] [Ask a Question]
### Wanted end result.

I believe library should offer/utilize sequences to some extent. It's too much to process all files using standard lists.

### What you tried so far.

I've tried to run architectural tests on quite big module, and it throws OOM.

**Comment by andriishpek-sigma** [comment]
I think OOM question duplicates #896. Nevertheless, I'm curious about Sequence usage

**Comment by igorwojda** [comment]
Sequences may improve performance a bit, but some checks require Konsist to parse entire code base and this is the root cause of OOM.

**Reply by andriishpek-sigma** [reply]
Yeah, having played around with different features I came across with resolving parents. I think this is especially the one you're talking about. Maybe sequences are obsolete after all.
Speaking of which, `KoParentProviderCore.parents()` and all related methods run extremely long comparing to all other filtering methods. TBH I haven't dig in, but it would be really nice to investigate ways to optimize this piece of logic. Let me know if you prefer starting a different discussion

**Reply by eygraber** [reply]
I think sequences could still be a pretty big help. I'm just getting started with Konsist, and it seems like there's a lot of filtering and mapping over large lists, which means they keep getting copied. Sequences would prevent a lot of that.
